packages/app/src/specs/banners/TrackedBanner.vue

```diff
@@ -8,9 +8,36 @@
   </Alert>
 </template>
 
+<script lang="ts">
+import { CohortConfig, useCohorts } from '@packages/frontend-shared/src/composables/useCohorts'
+
+export type CohortOption = {
+  cohort: string
+  value: string
+}
+
+export const getOptionForCohort = (name: string, options: CohortOption[]) => {
+  const cohortConfig: CohortConfig = {
+    name,
+    cohorts: options.map((option) => option.cohort),
+  }
+
+  const cohortSelected = useCohorts(cohortConfig)
```

Comment:
This feels error-prone - using a composable within a plain JS function will blow up if somebody happens to try use it outside a component (in a unit test, for example). Worth extracting this into the `useCohorts` composable so all the cohort logic is in one place?

Reply:
I wonder if there's a race condition here? Since `useCohorts` makes an API call, it's actually async - so this could potentially be undefined, then the API call finishes, then this would re-compute.

As an aside, surprisingly, this would not blow up in a unit test - Vue's reactivity is decoupled from the renderer, so you can actually use it outside the context of Vue - it's really "just javascript" (except it's not really clear what you'd want to use it outside of Vue for, I guess, except for writing a unit test - at that point, I'd argue you extract the logic out of the composable, and unit test that).

Reply:
Moved this logic into the composable

Reply:
Moved this logic into the composable